tools/compatibility/jersey/src/main/java/org/projectnessie/tools/compatibility/jersey/JerseyServer.java

```diff
@@ -0,0 +1,145 @@
+import java.net.URI;
+import java.util.function.Consumer;
+import java.util.function.Supplier;
+import javax.enterprise.inject.spi.Extension;
+import javax.ws.rs.core.Application;
+import org.glassfish.jersey.message.DeflateEncoder;
+import org.glassfish.jersey.message.GZipEncoder;
+import org.glassfish.jersey.server.ResourceConfig;
+import org.glassfish.jersey.server.filter.EncodingFilter;
+import org.glassfish.jersey.test.JerseyTest;
+import org.glassfish.jersey.test.TestProperties;
+import org.jboss.weld.environment.se.Weld;
+import org.projectnessie.services.authz.AbstractBatchAccessChecker;
+import org.projectnessie.services.authz.AccessContext;
+import org.projectnessie.services.authz.BatchAccessChecker;
+import org.projectnessie.services.impl.ConfigApiImpl;
+import org.projectnessie.services.impl.TreeApiImpl;
+import org.projectnessie.services.rest.ContentKeyParamConverterProvider;
+import org.projectnessie.services.rest.InstantParamConverterProvider;
+import org.projectnessie.services.rest.NessieExceptionMapper;
+import org.projectnessie.services.rest.NessieJaxRsJsonMappingExceptionMapper;
+import org.projectnessie.services.rest.NessieJaxRsJsonParseExceptionMapper;
+import org.projectnessie.services.rest.RestConfigResource;
+import org.projectnessie.services.rest.RestContentResource;
+import org.projectnessie.services.rest.RestTreeResource;
+import org.projectnessie.services.rest.ValidationExceptionMapper;
+import org.projectnessie.versioned.persist.adapter.DatabaseAdapter;
+
+/**
+ * Jersey server implementation, which uses reflection to get Nessie related classes.
+ *
+ * <p>This class and its companion classes in this package are used from {@code
+ * org.projectnessie.tools.compatibility.internal.OldNessieServer} via {@code
+ * org.projectnessie.tools.compatibility.internal.JerseyForOldServerClassLoader} and have access to
+ * an old Nessie server's class path.
+ */
+public class JerseyServer implements AutoCloseable {
+  private final Weld weld;
+  private final JerseyTest jerseyTest;
+
+  public JerseyServer(Supplier<DatabaseAdapter> databaseAdapterSupplier) throws Exception {
+    weld = new Weld();
+    // Let Weld scan all the resources to discover injection points and dependencies
+    weld.addPackages(true, TreeApiImpl.class);
+    // Inject external beans
+    weld.addExtension(new ServerConfigExtension());
+    weld.addExtension(PersistVersionStoreExtension.forDatabaseAdapter(databaseAdapterSupplier));
+    weld.addExtension(authzExtension());
+    weld.initialize();
+
+    jerseyTest =
+        new JerseyTest() {
+          @Override
+          protected Application configure() {
+            ResourceConfig config = new ResourceConfig();
+            config.register(RestConfigResource.class);
+            config.register(RestTreeResource.class);
+            config.register(RestContentResource.class);
+            withClass("org.projectnessie.services.rest.RestDiffResource", config::register);
```

Comment:
just thinking whether it would make sense to make it clearer that these classes are `new` classes that aren't available in the old version (whatever old means)

Reply:
I'm thinking about separating those `new` classes out, something like:

```
private List<Class<?>> classesSinceX() {
    return Arrays.asList(withClass("org.projectnessie.services.rest.RestDiffResource"),withClass("org.projectnessie.services.rest.RestRefLogResource"), ...);
  }
```

and then doing something like `classesSinceX().forEach(config::register);`

Reply:
Not a fan of `classesSinceX` (and X being a version number) - that would make the code here very very version specific (it's already a PITA that that module's needed due to class loader isolation requirements).

The alternative to this module would have been a lot of reflection based code - that's an even more a PITA.